### PR TITLE
PSP-2300 update organization - BACKEND

### PR DESCRIPTION
**Pull request by asanchezr**
This is the backend portion of UPDATE ORGANIZATION. As such, there is no UI (currently) to capture updated data for an organization. However I tested the API endpoints with Postman and they return the JSON data that is expected.

I have included a small sample of backend unit tests in this PR. More tests will come in either this or subsequent PRs.

### CODE REVIEW

File: backend/dal/Repositories/OrganizationRepository.cs [hunk 1]
```diff
@@ -93,10 +101,36 @@ public PimsOrganization Add(PimsOrganization organization, bool userOverride)
...
+            this.User.ThrowIfNotAuthorized(Permissions.ContactEdit);
+
+            var orgId = organization.Id;
+            var existingOrganization = this.Context.PimsOrganizations.Where(o => o.OrganizationId == orgId).FirstOrDefault()
+                 ?? throw new KeyNotFoundException();
+
+            // update main entity - PimsOrganization
+            Context.Entry(existingOrganization).CurrentValues.SetValues(organization);
+
+            // TODO: Verify these are working as expected!!!
```
Comment (asanchezr) [inline backend/dal/Repositories/OrganizationRepository.cs:125]: I will remove this TODO when the full end-to-end changes are implemented (including frontend forms)

File: backend/api/Areas/Organizations/Mapping/Organization/AddressMap.cs [hunk 2]
```diff
@@ -1,20 +1,42 @@
...
 namespace Pims.Api.Areas.Organizations.Mapping.Organization
 {
     public class AddressMap : IRegister
     {
         public void Register(TypeAdapterConfig config)
         {
-            config.NewConfig<Pims.Api.Models.Contact.AddressModel, Entity.PimsOrganizationAddress>()
-                .Map(dest => dest.AddressId, src => src.Id)
+            config.NewConfig<Entity.PimsOrganizationAddress, Model.OrganizationAddressModel>()
+                            .Map(dest => dest.Id, src => src.AddressId)
```
Comment (FuriousLlama) [inline backend/api/Areas/Organizations/Mapping/Organization/AddressMap.cs:13]: nit: indentation looks off

File: backend/api/Areas/Organizations/Mapping/Organization/AddressMap.cs [hunk 3]
```diff
@@ -1,20 +1,42 @@
...
+                            .Map(dest => dest.RowVersion, src => src.Address.ConcurrencyControlNumber)
```
[the remark this answers is not in the document]
Comment (asanchezr) [reply backend/api/Areas/Organizations/Mapping/Organization/AddressMap.cs:14]: AddressModel does not inherit from BaseAppModel (but could). Will make the change

File: backend/api/Areas/Organizations/Mapping/Organization/AddressMap.cs (hunk 2 above)
Comment (asanchezr) [reply backend/api/Areas/Organizations/Mapping/Organization/AddressMap.cs:13]: good catch

**Pull request merged**